pandas/io/formats/printing.py

```diff
@@ -381,7 +397,11 @@ def best_len(values):
         summary, line = _extend_line(summary, line, tail[-1],
                                      display_width - 2, space2)
         summary += line
-        summary += '],'
+
+        # right now close is either '' or ', '
+        # Now we want to include the ']', but not the maybe space.
```

Comment:
why is this needed?

Reply:
Without it, you'd get something like

```python
In [2]: pd.core.arrays.period_array(['2000', '2001'], freq='D')
Out[2]:
<PeriodArray>
['2000-01-01', '2001-01-01'], 
Length: 2, dtype: period[D]
```

(notice the trailing coma after the ending `]` in the data section.


That's what we want for index classes:

```python
In [3]: pd.Index([1, 2, 3])
Out[3]: Int64Index([1, 2, 3], dtype='int64')
```

but not for EAs since we don't know if they're valid code:

Reply:
i think this is a very strange repr 
where is this based off of? why not use parens?
it’s also multi line 

this is reinventing the wheel again compared to what we do for index; and somewhat arbitrary

repr is very important for consistency and this is off

i would avoid the special casing and have it look a whole lot more like what we have now

Reply:
also none of this looks tested (meaning the special casing and so on)

Reply:
the valid code argument does not work here either

we already have not valid code in repr eg MultiIndex and IntervalIndex

it’s very hard to guarantee this 

but using parens and no angle brackets and commas between items would be a major improvement

Reply:
> also none of this looks tested (meaning the special casing and so on)
 
What special casing? I have 100% coverage for the diff when running it on the base repr tests.

Reply:
Why's that? Don't we want to reuse the common formatting code?

Reply:
so another difference this is highliting is that EA have the attributes on another line, while the Index does not (as they are args).